sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala

```diff
@@ -79,6 +79,11 @@ private[sql] case class InMemoryTableScanExec(
 
     case IsNull(a: Attribute) => statsFor(a).nullCount > 0
     case IsNotNull(a: Attribute) => statsFor(a).count - statsFor(a).nullCount > 0
+
+    case In(a: AttributeReference, list: Seq[Expression])
+      if list.length <= inMemoryPartitionPruningMaxInSize =>
+      list.map(l => statsFor(a).lowerBound <= l.asInstanceOf[Literal] &&
```

Comment:
Oh, right. I miss that. I'll fix that by checking.
Thank you for review, @cloud-fan !